### PR TITLE
Ensure readability of MySQL and MariaDB config override

### LINKED ISSUE #914: Permissions error with config mounted into MySQL container
```
11:42:05.293 [Test worker] ERROR 🐳 [mysql:5.7.22] - Container log output (if any) will follow:
11:42:05.304 [tc-okhttp-stream-474827747] INFO 🐳 [mysql:5.7.22] - STDERR: 
11:42:05.305 [tc-okhttp-stream-474827747] INFO 🐳 [mysql:5.7.22] - STDERR: ERROR: mysqld failed while attempting to check config
11:42:05.305 [tc-okhttp-stream-474827747] INFO 🐳 [mysql:5.7.22] - STDERR: command was: "mysqld --verbose --help"
11:42:05.305 [tc-okhttp-stream-474827747] INFO 🐳 [mysql:5.7.22] - STDERR: 
11:42:05.306 [tc-okhttp-stream-474827747] INFO 🐳 [mysql:5.7.22] - STDERR: mysqld: Can't read dir of '/etc/mysql/conf.d/' (Errcode: 13 - Permission denied)
11:42:05.306 [tc-okhttp-stream-474827747] INFO 🐳 [mysql:5.7.22] - STDERR: mysqld: [ERROR] Fatal error in defaults handling. Program aborted!
```

We see this in the log (look on the null):
`
Cmd: com.github.dockerjava.core.command.CopyArchiveToContainerCmdImpl@2c8c6bb[cp ,<null>, ,6ca1ea3ce4b2582536fac69ceeb991670930b1ac65a6b2206c1ab88f3c990b65,:,/etc/mysql/]
11:41:45.064 [Test worker] INFO 🐳 [mysql:5.7.22] - Starting container with ID: 6ca1ea3ce4b2582536fac69ceeb991670930b1ac65a6b2206c1ab88f3c990b65
`

If we override the MySQLContainer class and just rmove this line from the configure function:
`
optionallyMapResourceParameterAsVolume(MY_CNF_CONFIG_OVERRIDE_PARAM_NAME, "/etc/mysql/conf.d", "mysql-default-conf");
`

Everything is working. Looks like it is copying invalid file to the container

### LINKED ISSUE #4952: JdbcDatabaseContainer#optionallyMapResourceParameterAsVolume should set explicit file permissions
I have a restrictive umask on my system that only allows me to read the files I create. Recently I tried to start a mysql database with Testcontainers. Testcontainers [maps a classpath resource into the config directory](https://github.com/testcontainers/testcontainers-java/blob/f9538d16b7f636420cc269ce15fd6b5742be278c/modules/mysql/src/main/java/org/testcontainers/containers/MySQLContainer.java#L64) of the container. To do this, it first extracts the contents to a temporary location and then uses whatever permissions this temp file has. Due to my umask, the temp file is only readable by me, but not by the user inside the mysql container, resulting in `mysqld: Can't read dir of '/etc/mysql/conf.d/' (Errcode: 13 - Permission denied)`

I'd suggest that the config files be created with permissions that allow it to be read by the container.

### PR DESCRIPTION
**Pull request by famod**
Fixes #914, fixes #4952, supersedes (and heavily based on) #5401.

Most credits to @driverpt!

### CODE REVIEW

**Comment by driverpt** [comment]
The credit is all yours my friend!

File: modules/mysql/src/test/java/org/testcontainers/junit/mysql/SimpleMySQLTest.java [hunk 1]
```diff
@@ -234,8 +240,43 @@ public void testWithAdditionalUrlParamInJdbcUrl() {
         }
     }
 
+    @Test
+    public void testWithOnlyUserReadableCustomIniFile() throws Exception {
```
Comment (eddumelendez) [inline modules/mysql/src/test/java/org/testcontainers/junit/mysql/SimpleMySQLTest.java:244]: Should I be able to reproduce this setting umask with 027, right?

**Review by eddumelendez — COMMENTED** [review]
I've left just one question. Just want to make sure the test is needed due to, previously, `testMySQLWithCustomIniFile` used to fail for `MariaDBContainer` and `MySQLContainer` and now it passes when running on Codespaces.

File: modules/mysql/src/test/java/org/testcontainers/junit/mysql/SimpleMySQLTest.java (hunk 1 above)
Comment (famod) [reply modules/mysql/src/test/java/org/testcontainers/junit/mysql/SimpleMySQLTest.java:244]: Yeah, the test kinda only an "emulation" but I tested my real world setup with this fix using `umask 0027`.

Comment (famod) [reply modules/mysql/src/test/java/org/testcontainers/junit/mysql/SimpleMySQLTest.java:244]: PS: I also did run this new test before and after the fix; before the fix it failed and now it passes.

**Review by eddumelendez — APPROVED** [review]

**Comment by eddumelendez** [comment]
Thanks for your contribution, @famod ! This is now merged in `main` branch and it will be part of the next release.

**Pull request merged**